### ISSUE TITLE
package import from a module broken since upgrading from 0.4.1 to 0.4.3-beta.1

### ISSUE THREAD

**Issue by mpvl** [cue-lang/cue#1599]
### Discussed in https://github.com/cue-lang/cue/discussions/1596

<div type='discussions-op-text'>

<sup>Originally posted by **jbcpollak** March 25, 2022</sup>
Hello, I just tried upgrading from 0.4.1 to 0.4.3-beta.1 and my module that used to work is now broken for `eval`, but seems to work for `vet`:

Here is the module:
```
-- config/cue.mod/module.cue --
module: "example.com/config"

-- config/components/widget.cue --
package widget

#Widget: {
        model: string
        explicit: "literal string"
}

-- config/config.cue --
package config

import (
	"example.com/config/components:widget"
)

w: widget.#Widget

-- config/instance.json --
{
  "w": {
    "model": "retroencabulator"
  }
}
```

`cue vet` seems to be fine:
```
❯ cue vet . instance.json
> echo $?
0
```

but `cue eval` outputs an error:

```
❯ cue eval . instance.json
w: cannot find package "example.com/config/components:widget":
    ./config.cue:7:4
```
</div>

**Comment by mpvl** [comment]
commit 195cdf7e74e4b633419f2fa2a605bd932f9889aa

Error was introduced by:

internal/core/runtime: don't share import cache
    
    This was always supposed to be this way, but dropped off
    the radar as a thing to do.
    
    Fixes #1424
    Fixes #1418
    Issue #1414

**Comment by mpvl** [comment]
Fixed by: https://review.gerrithub.io/c/cue-lang/cue/+/535433 / aca00a068569f36b95c264e216b5d22ff9759a91
